endpoint.go

```diff
@@ -122,17 +121,39 @@ type InEndpoint struct {
 	*endpoint
 }
 
-// Read reads data from an IN endpoint.
+// Read reads data from an IN endpoint. Read returns number of bytes obtained
+// from the endpoint. Read may return non-zero length even if
+// the returned error is not nil (partial read).
 func (e *InEndpoint) Read(buf []byte) (int, error) {
-	return e.transfer(buf)
+	return e.transfer(context.Background(), buf)
+}
+
+// ReadContext reads data from an IN endpoint. Read returns number of bytes
+// obtained from the endpoint. Read may return non-zero length even if the
+// returned error is not nil (partial read).
+// The passed context can be used to control the cancellation of the read. If
+// the context is cancelled, ReadContext will return a TransferTimedOut error.
+func (e *InEndpoint) ReadContext(ctx context.Context, buf []byte) (int, error) {
+	return e.transfer(ctx, buf)
 }
 
 // OutEndpoint represents an OUT endpoint open for transfer.
 type OutEndpoint struct {
 	*endpoint
 }
 
-// Write writes data to an OUT endpoint.
+// Write writes data to an OUT endpoint. Write returns number of bytes comitted
+// to the endpoint. Write may return non-zero length even if the returned error
+// is not nil (partial write).
 func (e *OutEndpoint) Write(buf []byte) (int, error) {
-	return e.transfer(buf)
+	return e.transfer(context.Background(), buf)
+}
+
+// WriteContext writes data to an OUT endpoint. Write returns number of bytes
```

Comment:
s/Write/WriteContext, also in the next line

Reply:
done